### PR TITLE
Ma/tesseract edge cases

### PR DESCRIPTION
**Pull request by Darth-Sand**
Created a testing file for ocr edge cases in the text_extraction test folder

### CODE REVIEW

File: src/text_extraction/test/test_ocr.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                passed_chars.append(ground_truth[char])
+            else:
+                declined_chars.append(ground_truth[char])
+
+        chars_evaluated = (len(passed_chars) > 0 or len(declined_chars) > 0)
+
+        print("passed characters:", passed_chars, "\n")
+        print("declined characters:", declined_chars, "\n")
+        print("passed:", len(passed_chars) ,"|","declined:", len(declined_chars), "\n")
+        print("the extracted text: \n", text)
```
Comment (JTC2000Official) [inline src/text_extraction/test/test_ocr.py]: Ingen print statements i testene

File: src/spell_checking/test/test_integration_spellchecker.py [hunk 2]
```diff
@@ -10,22 +11,32 @@ class SpellcheckerIntegrationTests(unittest.TestCase):
...
+        ground_truth = "src/spell_checking/test/test_files/expected.txt"
         text_extractor = TextExtractor()
         text_extractor.out_dir = "/watched/spell_checking/"
         spellchecker = SpellChecker("src/spell_checking/wordList.txt")
         spellchecker.out_dir = "/watched/output"
-        with open("src/spell_checking/test/expected.txt", 'r', encoding="utf-8") as expected_text:
+        #register expected.txt as a list
+        with open(ground_truth, 'r', encoding="utf-8") as expected_text:
             expected_text = expected_text.read().lower().split()
             print(f'Expected text: {expected_text}')
```
Comment (JTC2000Official) [inline src/spell_checking/test/test_integration_spellchecker.py]: Fjern print statement

File: src/spell_checking/test/test_integration_spellchecker.py [hunk 3]
```diff
@@ -10,22 +11,32 @@ class SpellcheckerIntegrationTests(unittest.TestCase):
...
+        shutil.copy(src, dst)
 
         #Act
+        #if a txt file has already been processed do not run extraction again
         if not os.path.exists("/watched/output/Test_File.txt"):
             text_extractor.read("src/spell_checking/test/Test_File.jpg")
             spellchecker.handle_files("/watched/spell_checking/Test_File.txt")
         with open("/watched/output/Test_File.txt", 'r', encoding="utf-8") as output:
             output = output.read().lower().split()
             print(f'Spellchecked text: {output}')
```
Comment (JTC2000Official) [inline src/spell_checking/test/test_integration_spellchecker.py]: Fjern print

File: src/text_extraction/test/test_ocr.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        #Assert
+        self.assertTrue(status, "Nothing extracted")
+        self.assertGreater(chars_evaluated, 0, "characters have not been evaluated")
```
Comment (JTC2000Official) [inline src/text_extraction/test/test_ocr.py]:
```python
self.assertGreater([...])
```
Hvorfor greater, når du arbejder med bool?

File: src/text_extraction/test/test_check_text.py [hunk 5]
```diff
@@ -14,12 +17,24 @@ def test_compare_text(self):
...
+        #copies test file since text extractor deletes files once processed
+        shutil.copy(src, dst)
 
         #Act
+        text_extractor.read("src/text_extraction/test/extracted.png")
+        #convert expected and extracted texts into lists
+        extracted_text = get_word_stream("src/text_extraction/test/extracted.txt")
+        expected_text = get_word_stream("src/text_extraction/test/test_files/expected.txt")
+        #compares extracted text to the expected text to see if extraction was a success
         result = extracted_text == expected_text
```
Comment (JTC2000Official) [inline src/text_extraction/test/test_check_text.py]: Fjern

File: src/text_extraction/test/test_check_text.py [hunk 6]
```diff
@@ -14,12 +17,24 @@ def test_compare_text(self):
...
+        #deletes the testing file that was copied over
+        if os.path.exists("src/text_extraction/test/extracted.png"):
+            os.remove("src/text_extraction/test/extracted.png")
 
         #Assert
-        self.assertTrue(result, 'the results is false thus the extracted is not correct')
+        self.assertTrue(result, 'the results is false thus the extracted text is not correct')
```
Comment (JTC2000Official) [inline src/text_extraction/test/test_check_text.py]: Brug:
```python
self.assertEqual(extracted_text, expected_text)
```

File: src/spell_checking/test/test_integration_spellchecker.py [hunk 7]
```diff
@@ -10,22 +11,32 @@ class SpellcheckerIntegrationTests(unittest.TestCase):
...
             status = bool(output == expected_text)
```
Comment (JTC2000Official) [inline src/spell_checking/test/test_integration_spellchecker.py]: Fjern

File: src/spell_checking/test/test_integration_spellchecker.py [hunk 8]
```diff
@@ -10,22 +11,32 @@ class SpellcheckerIntegrationTests(unittest.TestCase):
...
+        #deletes the testing file that was copied over
+        if os.path.exists("src/spell_checking/test/Test_File.jpg"):
+            os.remove("src/spell_checking/test/Test_File.jpg")
 
         #Assert
         self.assertTrue(status, "The text was not extracted correctly")
```
Comment (JTC2000Official) [inline src/spell_checking/test/test_integration_spellchecker.py]: Slet og Brug:
```python
self.assertEqual(output, expected_text)
```

**Review by JTC2000Official — CHANGES_REQUESTED** [review]
Se kommentarer

File: src/text_extraction/test/test_ocr.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+    print("passed input: ", passed_chars, "\n")
+    print("declined input: ", declined_chars, "\n")
+    print("passed:", len(passed_chars), "|" + "declined:", len(declined_chars), "\n")
+    print("the expected text: \n" + "------------------\n", raw_text, "\n")
+    print("the extracted text: \n" + "-------------------\n" + text)
+
+if __name__ == "__main__":
+    test_edge_cases()
+
```
Comment (JTC2000Official) [inline src/text_extraction/test/test_ocr.py]: Skal det her anses som en unit-/integrationtest?
Hvis nej, hvad er begrundelse for at placerer under test mappen etc.

**Review by JTC2000Official — COMMENTED** [review]
Kan ikke se at alle krav opfyldes, se følgende:
"Test pytesseract med dokumenter som har svære tegn at læse, samt tegn der er lidt specielle for at se hvad den kan klares"

Mangler test af specielle tegn

**Review by JTC2000Official — DISMISSED** [review]

**Review by JTC2000Official — APPROVED** [review]
LGTM
